src/jsc/jsc_value.hpp

```diff
@@ -41,6 +41,37 @@ static inline bool is_object_of_type(JSContextRef ctx, JSValueRef value, jsc::St
     return result;
 }
 
+// Look up and verify against Realm._bson.<type>
+static inline bool is_object_of_bson_type(JSContextRef ctx, JSValueRef value, jsc::String type) {
+    static const jsc::String realm_name = "Realm";
+    static const jsc::String bson_name = "_bson";
```

Comment:
should this be `realmConstructor._ObjectId = realmConstructor._bson.ObjectId;`

Reply:
ignore

Reply:
It would save us a lookup